ported/omb/rdm_bw.c

```diff
@@ -0,0 +1,480 @@
+	ret = fi_av_insert(av, addrs, numprocs, fi_addrs, 0, &fi_ctx_av);
+	if (ret != numprocs) {
+		FT_PRINTERR("fi_av_insert", ret);
+		return ret;
+	}
+
+	free(addr);
+
+	return 0;
+}
+
+int main(int argc, char *argv[])
+{
+	int i, j, peer;
+	int size, align_size;
+	char *s_buf, *r_buf;
+	uint64_t t_start = 0, t_end = 0, t = 0;
+	int op, ret;
+	buf_desc_t lbuf_desc;
+	ssize_t fi_rc;
+
+	FT_Init(&argc, &argv);
+	FT_Rank(&myid);
+	FT_Job_size(&numprocs);
+
+	hints = fi_allocinfo();
+	if (!hints)
+		return -1;
+
+	while ((op = getopt(argc, argv, "h" INFO_OPTS)) != -1) {
+		switch (op) {
+		default:
+			ft_parseinfo(op, optarg, hints);
+			break;
+		case '?':
+		case 'h':
+			print_usage();
+			return EXIT_FAILURE;
+		}
+	}
+
+	hints->ep_attr->type	= FI_EP_RDM;
+	hints->caps		= FI_MSG | FI_DIRECTED_RECV | FI_RMA;
+	hints->mode		= FI_CONTEXT | FI_LOCAL_MR;
+	hints->domain_attr->mr_mode = FI_MR_BASIC;
+
+	if (numprocs != 2) {
+		if (myid == 0) {
+			fprintf(stderr, "This test requires exactly two processes\n");
+		}
+		FT_Finalize();
+		return -1;
+	}
+
+	/* Fabric initialization */
+	ret = init_fabric();
+	if (ret) {
+		fprintf(stderr, "Problem in fabric initialization\n");
+		return ret;
+	}
+
+	ret = init_av();
+	if (ret) {
+		fprintf(stderr, "Problem in AV initialization\n");
+		return ret;
+	}
+
+	/* Data initialization */
+	align_size = getpagesize();
+	assert(align_size <= MAX_ALIGNMENT);
+
+	s_buf = (char *) (((unsigned long) s_buf_original + (align_size - 1)) /
+				align_size * align_size);
+	r_buf = (char *) (((unsigned long) r_buf_original + (align_size - 1)) /
+				align_size * align_size);
+
+	ret = fi_mr_reg(dom, r_buf, MYBUFSIZE, FI_REMOTE_WRITE, 0, 0, 0, &r_mr, NULL);
+	if (ret) {
+		FT_PRINTERR("fi_mr_reg", ret);
+		return -1;
+	}
+
+	lbuf_desc.addr = (uint64_t)r_buf;
+	lbuf_desc.key = fi_mr_key(r_mr);
+
+	rbuf_descs = (buf_desc_t *)malloc(numprocs * sizeof(buf_desc_t));
+
+	/* Distribute memory keys */
+	FT_Allgather(&lbuf_desc, sizeof(lbuf_desc), rbuf_descs);
+
+	ret = fi_mr_reg(dom, s_buf, MYBUFSIZE, FI_WRITE, 0, 0, 0, &l_mr, NULL);
+	if (ret) {
+		FT_PRINTERR("fi_mr_reg", ret);
+		return -1;
+	}
+
+	if (myid == 0) {
+		fprintf(stdout, HEADER);
+		fprintf(stdout, "%-*s%*s\n", 10, "# Size", FIELD_WIDTH,
+				"Bandwidth (MB/s)");
+		fflush(stdout);
+	}
+
+	/* Bandwidth test */
+	for (size = 1; size <= MAX_MSG_SIZE; size *= 2) {
+		/* touch the data */
+		for (i = 0; i < size; i++) {
+			s_buf[i] = 'a';
+			r_buf[i] = 'b';
+		}
+
+		if (size > large_message_size) {
+			loop = loop_large;
+			skip = skip_large;
+			window_size = window_size_large;
+		}
+
+		FT_Barrier();
+
+		if (myid == 0) {
+			peer = 1;
+
+			for (i = 0; i < loop + skip; i++) {
+				if (i == skip) {
+					t_start = get_time_usec();
+				}
+
+				for (j = 0; j < window_size; j++) {
+					fi_rc = fi_write(ep, s_buf, size, l_mr,
+							fi_addrs[peer],
+							rbuf_descs[peer].addr,
+							rbuf_descs[peer].key,
+							(void *)(intptr_t)j);
```

Comment:
I followed the earlier discussion about using tagged interfaces. But, will it make sense to rename this to put_bw (or write_bw) since we use RMA write operations in this benchmark?

Also, should we rename the latency test to rdm_send_lat? Later if we add the tagged interfaces we can name it rdm_tsend_lat (Just a thought).